src/blockchain/log-processors/order-canceled.ts

```diff
@@ -20,17 +23,65 @@ interface MarketNumOutcomes {
 
 export async function processOrderCanceledLog(augur: Augur, log: FormattedEventLog) {
   return async (db: Knex) => {
+    const { tokensEscrowed }: Pick<OrdersRow<BigNumber>, "tokensEscrowed"> = await db
+      .first("orders.tokensEscrowed")
+      .from("orders")
+      .where({ orderId: log.orderId })
+      .orderByRaw(`"blockNumber" DESC, "logIndex" DESC`);
+    if (!tokensEscrowed) throw new Error(`order id not found: ${{ orderId: log.orderId }}`);
+
+    const prevProfitLoss: undefined | ProfitLossTimeseriesRow<BigNumber> = await db
+      .first("wcl_profit_loss_timeseries.*")
+      .from("wcl_profit_loss_timeseries")
+      .join("orders", function(this: any) {
+        this.on(function(this: any) {
+          this.on("wcl_profit_loss_timeseries.marketId", "=", "orders.marketId");
+          this.andOn("wcl_profit_loss_timeseries.outcome", "=", "wcl_profit_loss_timeseries.outcome");
+        });
+      })
+      .whereRaw("wcl_profit_loss_timeseries.account = orders.orderCreator")
+      .where({ orderId: log.orderId })
+      .orderByRaw(`"blockNumber" DESC, "logIndex" DESC, "wcl_profit_loss_timeseries.rowid" DESC`);
+
+    if (!prevProfitLoss) throw new Error(`expected to find previous wcl_profit_loss_timeseries on order cancel orderId=${log.orderId}`);
+
     const orderTypeLabel = log.orderType === "0" ? "buy" : "sell";
     await db.from("orders").where("orderId", log.orderId).update({ orderState: OrderState.CANCELED });
-    await  db.into("orders_canceled").insert({ orderId: log.orderId, transactionHash: log.transactionHash, logIndex: log.logIndex, blockNumber: log.blockNumber });
-    const ordersRow: MarketIDAndOutcomeAndPrice = await  db.first("marketId", "outcome", "price", "sharesEscrowed", "orderCreator").from("orders").where("orderId", log.orderId);
+    await db.into("orders_canceled").insert({ orderId: log.orderId, transactionHash: log.transactionHash, logIndex: log.logIndex, blockNumber: log.blockNumber });
+
+    const nextFrozenFunds: FrozenFunds = getFrozenFundsAfterEventForOneOutcome({
+      frozenFundsBeforeEvent: prevProfitLoss,
```

Comment:
Should `frozenFundsBeforeEvent` be labeled `profitLossBeforeEvent` maybe?

Reply:
I think maybe it is a different thing, `frozenFundsBeforeEvent` is a member of `FrozenFundsParams` and `prevProfitLoss` is being downcast into that type